### PR TITLE
Update ansible template, readme, and script to bring in line with Ansible Galaxy

### PR DESCRIPTION
**Pull request by redhatrises**
- Align role name to `ansible_role_<profile_name>`
- Add `role_name` to metadata to allow simpler short role naming
- Add new capability to tag releases 
- Remove XCCDF references in readme. This doesn't make sense for an ansible role
- Remove references to "Playbook" unless used in the context of calling an ansible role
- Add product whitelist to remove hardcoded `rhel7`
- Add capability to pass token with arg option
- Update Readme and metadata only with non-role specific (e.g. role tags) changes
- Map ssg ansible role yaml files to `ansible_role_<profile_name>` convention

### CODE REVIEW

File: utils/ansible_galaxy_readme_template.md [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 @DESCRIPTION@
 
+The tasks that are used in this role are generated using OpenSCAP.
+See the OpenSCAP project for more details at [https://github.com/OpenSCAP/openscap](https://github.com/OpenSCAP/openscap)
```
Comment (shawndwells) [inline utils/ansible_galaxy_readme_template.md]: Should ComplianceAsCode be mentioned here instead?

e.g. ``Tasks used in this role are generated from the ComplianceAsCode Project. For more details, see https://github.com/ComplianceAsCode/content``

File: utils/ansible_galaxy_readme_template.md [hunk 2]
```diff
@@ -18,17 +21,17 @@ Dependencies
 
 N/A
 
-Example Playbook
+Example Role Usage
```
Comment (shawndwells) [inline utils/ansible_galaxy_readme_template.md]: Thanks for doing this. Know it was annoying some people :)

File: utils/ansible_galaxy_readme_template.md [hunk 3]
```diff
@@ -18,17 +21,17 @@ Dependencies
...
 Run `ansible-galaxy install RedHatOfficial.@ROLE_NAME@` to
-download and install the role. Then you can use the following playbook snippet.
+download and install the role. Then, you can use the following playbook snippet to run the Ansible role:
 
     - hosts: all
       roles:
          - { role: RedHatOfficial.@ROLE_NAME@ }
 
-Then first check the playbook using (on the localhost):
+Then, check the playbook using (on the localhost) the following example:
```
Comment (shawndwells) [inline utils/ansible_galaxy_readme_template.md]:
```suggestion
Next, check the playbook using (on the localhost) the following example:
```

Comment (shawndwells) [reply utils/ansible_galaxy_readme_template.md]: (because ``Then,...`` was used in the last paragraph.

Comment (redhatrises) [reply utils/ansible_galaxy_readme_template.md]: @shawndwells fixed

File: utils/ansible_galaxy_readme_template.md (hunk 1 above)
Comment (redhatrises) [reply utils/ansible_galaxy_readme_template.md]: @shawndwells I added both. People need to understand that the tasks are generated using OpenSCAP, but that ComplianceAsCode is where to submit fixes.

File: utils/upload_ansible_roles_to_galaxy.py [hunk 4]
```diff
@@ -362,10 +445,20 @@ def locally_clone_and_init_repositories(organization, repo_list):
 def main():
     args = parse_args()
 
-    all_role_whitelist = {"rhel7-role-%s" % p for p in PROFILE_WHITELIST}
+    all_product_whitelist = {"%s" % p for p in PRODUCT_WHITELIST}
```
Comment (jan-cerny) [inline utils/upload_ansible_roles_to_galaxy.py]: `all_product_whitelist` is now a set that contains same elements as `PRODUCT_WHITELIST`. Please remove this.

File: utils/upload_ansible_roles_to_galaxy.py [hunk 5]
```diff
@@ -362,10 +445,20 @@ def locally_clone_and_init_repositories(organization, repo_list):
...
+    product_whitelist = set(all_product_whitelist)
+    if args.product:
+        selected_products = {"%s" % p for p in args.product}
```
Comment (jan-cerny) [inline utils/upload_ansible_roles_to_galaxy.py]: `selected_products = set(args.product)` will do the same job

File: utils/upload_ansible_roles_to_galaxy.py [hunk 6]
```diff
@@ -362,10 +445,20 @@ def locally_clone_and_init_repositories(organization, repo_list):
...
+
+    all_role_whitelist = []
+    for product in product_whitelist:
+        for p in PROFILE_WHITELIST:
+            all_role_whitelist.append("%s-role-%s" % (product, p))
+
     role_whitelist = set(all_role_whitelist)
     if args.profile:
-        selected_roles = {"rhel7-role-%s" % p for p in args.profile}
+        selected_roles = {"%s" % p for p in args.profile}
```
Comment (jan-cerny) [inline utils/upload_ansible_roles_to_galaxy.py]: `selected_roles = set(args.profile)` should do the same job, but it's more understandable.

Comment (jan-cerny) [reply utils/upload_ansible_roles_to_galaxy.py]: Let's assume I run the script with `--product rhel8 --profile ospp`. At this point, the variables contain:
```
role_whitelist = {'rhel8-role-nist-800-171-cui', 'rhel8-role-pci-dss', 'rhel8-role-ospp', 'rhel8-role-stig-rhel7-disa', 'rhel8-role-hipaa', 'rhel8-role-C2S', 'rhel8-role-cjis', 'rhel8-role-rht-ccp'}
selected_roles = {'ospp'} 
```
The intersection of those 2 sets is obviously an empty set, so after executing the line below the `role_whitelist` is an empty set. That means no role will be updated if user specifies `--profile`.

File: utils/upload_ansible_roles_to_galaxy.py (hunk 5 above)
Comment (redhatrises) [reply utils/upload_ansible_roles_to_galaxy.py]: @jan-cerny fixed

File: utils/upload_ansible_roles_to_galaxy.py (hunk 4 above)
Comment (redhatrises) [reply utils/upload_ansible_roles_to_galaxy.py]: @jan-cerny fixed

File: utils/upload_ansible_roles_to_galaxy.py (hunk 6 above)
Comment (redhatrises) [reply utils/upload_ansible_roles_to_galaxy.py]: @jan-cerny this should be working now.

**Comment by jan-cerny** [comment]
It works OK for me.

I have one more suggestion: People who use SSH key authentication would prefer if the repository is cloned using SSH instead of HTTPs here: https://github.com/ComplianceAsCode/content/blob/0cbce045d17ec1adf4d095b37893b1cf1ecc7654/utils/upload_ansible_roles_to_galaxy.py#L86
i.e. `git clone git@github.com:%s/%s`.
If HTTPs is used, it asks for username and passphrase which is annoying, and the authentication fails when people use 2FA. If the SSH is used it would work seamlessly.

**Comment by redhatrises** [comment]
> It works OK for me.
> 
> I have one more suggestion: People who use SSH key authentication would prefer if the repository is cloned using SSH instead of HTTPs here:
> 
> [content/utils/upload_ansible_roles_to_galaxy.py](https://github.com/ComplianceAsCode/content/blob/0cbce045d17ec1adf4d095b37893b1cf1ecc7654/utils/upload_ansible_roles_to_galaxy.py#L86)
> 
> Line 86 in [0cbce04](/ComplianceAsCode/content/commit/0cbce045d17ec1adf4d095b37893b1cf1ecc7654)
>  "git clone https://github.com/%s/%s" % (organization, repo)) 
> 
> i.e. `git clone git@github.com:%s/%s`.
> If HTTPs is used, it asks for username and passphrase which is annoying, and the authentication fails when people use 2FA. If the SSH is used it would work seamlessly.

Github recommend `https` over ssh. Also, there was numerous issues with script using the git version. For handling 2FA, using a github personal access token makes everything all easy and simple.

**Comment by jan-cerny** [comment]
@redhatrises Aha OK

**Pull request merged**